### DISCUSSION TITLE
Installed java jdk versus downloaded dependency

### DISCUSSION THREAD

**Discussion by JeremyCavanagh** [Homebrew/discussions#1431] [Everyday usage]
Hi All/Anyone,

I have a problem which causes some consternation for me it's probably no big deal but it bugs/confuses me.

Generally,  why don't brew formulae check the system to see if a suitable java jdk is already installed before downloading a dependency. 

A specific example is jenkins-lts which doesn't check and downloads another version of openjdk when I already have an appropriate  zulu jdk installed in /System/Java/JavaVirtualMachines other instances have caused much time wasting when a new jdk is installed in the homebrew ecosystem.

Regards
Jeremy

**Comment by gromgit** [comment]
> Generally, why don't brew formulae check the system to see if a suitable java jdk is already installed before downloading a dependency.

Because Homebrew formulae are tested against specific versions of Homebrew-provided dependencies.

Also, I wouldn't be surprised if Homebrew used to do so, but stopped because addressing ten thousand "it doesn't work" complaints from folks with arbitrary JVM versions in various stages of disrepair gets tiresome really quickly.
